### PR TITLE
Connections/debug and improvements

### PR DESCRIPTION
**Pull request by pv3nturi**
- DebugFilter to add come connections data with  endpoints.

- When release forceClose is performed now connection is returned only when closeFuture complete.

### CODE REVIEW

File: carapace-server/src/main/java/org/carapaceproxy/client/impl/EndpointConnectionImpl.java [hunk 1]
```diff
@@ -393,14 +394,19 @@ public void release(boolean close, RequestHandler clientSidePeerHandler, Runnabl
...
-    void destroy() {
+    void destroy(Runnable onCloseCompleted) {
         invalidate();
         if (!closed.compareAndSet(false, true)) {
             return;
 
         }
         // note: close is async, we are not waiting for real close
-        channelToEndpoint.close();
+        ChannelFuture closeFuture = channelToEndpoint.close();
```
Comment (nicoloboschi) [inline carapace-server/src/main/java/org/carapaceproxy/client/impl/EndpointConnectionImpl.java:404]: If we want to wait we have to call .get() after the close()

File: carapace-server/src/main/java/org/carapaceproxy/server/filters/DebugFilter.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+/**
+ * Adds a debug header.
+ *
+ * @author paolo.venturi
+ */
+public class DebugFilter extends BasicRequestFilter {
+
+    public static final String TYPE = "debug";
+    public static final String DEBUG_HEADER_NAME = "X-Carapace-Debug";
```
Comment (nicoloboschi) [inline carapace-server/src/main/java/org/carapaceproxy/server/filters/DebugFilter.java:35]: X-Carapace-CID or similar

**Review by nicoloboschi — CHANGES_REQUESTED** [review]
It looks good, I left two comments

File: carapace-server/src/main/java/org/carapaceproxy/server/filters/DebugFilter.java (hunk 2 above)
Comment (pv3nturi) [reply carapace-server/src/main/java/org/carapaceproxy/server/filters/DebugFilter.java:35]: I meant that as a list of debug info for future improvements (and low headers spam)

**Comment by pv3nturi** [comment]
Let's create a brand new branch with debug-header only.

**Pull request closed without merging**